### PR TITLE
Add GenerateSasUri

### LINKED ISSUE #16917: Implement GenerateSasUri pattern for SAS token generation
# Summary
Adopt the `GenerateSasUri` SAS token generation pattern adopted by storage clients.

See [this](https://github.com/Azure/azure-sdk-for-net/blob/ece88b87fd3b106df442c9bc2743f75fd4d592f7/sdk/storage/Azure.Storage.Queues/src/QueueClient.cs) for reference

### LINKED ISSUE #20481: [QUERY] Cannot understand how to remove REDACTED when using AzureEventSourceListener
**Query/Question**
I've read page https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md about activating logging.

I used:
```
using AzureEventSourceListener listener = AzureEventSourceListener.CreateConsoleLogger();
```
Output looks like:
```
Azure-Core: Request [a9113e9a-44fd-48f0-bb8f-45024850e3d9] GET http://127.0.0.1:10002/devstoreaccount1/aaaatest1()?$format=REDACTED&$filter=REDACTED
x-ms-version:REDACTED
DataServiceVersion:REDACTED
.....
```
I do not understand how to remove REDACTED and having full parameters visible. The page mentions `SecretClientOptions options` but it's not clear how to use it with  AzureEventSourceListener.CreateConsoleLogger 

**Environment:**
 - Name and version of the Library package used: Azure.Data.Tables - 12.0.0-beta.7

### LINKED ISSUE #20909: TableSasBuilder generates wrong signatures when given non-lowercase table names
Azure requires the table name to be lowercase in the signed string's canonical name and in the resulting tn parameter in the query string.  TableSasBuilder should be calling tableName.ToLowerInvariant() in the following two places:

https://github.com/Azure/azure-sdk-for-net/blob/d9de4404133151c351cf6ef9431b4dc21d4e3509/sdk/tables/Azure.Data.Tables/src/Sas/TableSasBuilder.cs#L215

https://github.com/Azure/azure-sdk-for-net/blob/d9de4404133151c351cf6ef9431b4dc21d4e3509/sdk/tables/Azure.Data.Tables/src/Sas/TableSasBuilder.cs#L259

### PR DESCRIPTION
**Pull request by christothes**
closes #16917
fixes #20481
fixes #20909

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -605,7 +605,7 @@ sdk/trafficmanager/Microsoft.Azure.Management.TrafficManager/            @tmsupp
 /sdk/tables/                                    @christothes
 
 # ServiceLabel: %Tables %Service Attention
-#/<NotInRepo>/            @klaaslanghout
+#/<NotInRepo>/            @sakash279, @sivethe, @ThomasWeiss, @PaulCheng
```
Comment (jsquire) [inline .github/CODEOWNERS:608]: I'll move this into the bot this afternoon.

File: sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs [hunk 2]
```diff
@@ -789,6 +825,52 @@ public virtual Response<TableServiceStatistics> GetStatistics(CancellationToken
...
+        /// Constructing a Service SAS</see>
+        /// </summary>
+        /// <param name="builder"> Used to generate a Shared Access Signature (SAS). </param>
+        /// <returns> A <see cref="TableAccountSasBuilder"/> on successfully deleting. </returns>
+        /// <remarks> An <see cref="Exception"/> will be thrown if a failure occurs. </remarks>
+        public virtual Uri GenerateSasUri(
+            TableAccountSasBuilder builder)
+        {
+            builder = builder ?? throw Errors.ArgumentNull(nameof(builder));
+            Argument.AssertNotNull(builder, nameof(builder));
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs:867]: I think this may be redundant to the line above...

**Review by jsquire — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs:867]: you're correct, I meant to delete the first line!

File: .github/CODEOWNERS (hunk 1 above)
Comment (jsquire) [reply .github/CODEOWNERS:608]: Done.

**Pull request merged**